context/UserInactivity.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from "react";
+import { AppState, AppStateStatus, NativeEventSubscription } from 'react-native';
+import { secureStorage } from "~/lib/secureStorage";
+import { INACTIVITY_THRESHOLD } from "~/lib/constants";
+import { useAppStore } from "~/lib/state/appStore";
+
+export const UserInactivityProvider = ({ children }: any) => {
+  const [appState, setAppState] = React.useState<AppStateStatus>(AppState.currentState);
+  const isSecurityEnabled = useAppStore((store) => store.isSecurityEnabled);
+
+  const handleAppStateChange = async (nextState: AppStateStatus) => {
+    if (appState === "active" && nextState.match(/inactive|background/)) {
+      const now = Date.now();
+      secureStorage.setItem("lastActiveTime", now.toString());
+    } else if (appState.match(/inactive|background/) && nextState === "active") {
+      const lastActiveTime = secureStorage.getItem("lastActiveTime");
+      if (lastActiveTime) {
+        const timeElapsed = Date.now() - parseInt(lastActiveTime, 10);
+        if (timeElapsed >= INACTIVITY_THRESHOLD) {
+          useAppStore.getState().setUnlocked(false)
+        }
+      }
+      await secureStorage.removeItem("lastActiveTime");
+    }
+    setAppState(nextState);
+  };
+
+  React.useEffect(() => {
+    let subscription: NativeEventSubscription
+    if (isSecurityEnabled) {
```

Comment:
I left an edgecase here, if the user keeps Alby Go in the background (without closing) and goes to settings and removes phone lock, and comes back to Go, then it just hangs in the Unlock screen because the security enabled is only checked during app open (hence if the user closes and opens it again they can access the app normally)

cc @reneaaron let me know if this is okay (did not add a check for everytime the app goes to bg and comes back that can solve the issue)

Reply:
So worst case the user would need to restart the app, right?

Reply:
Yup

Reply:
Guess that's fine for now. :+1: